src/libs/actions/PersistedRequests.ts

```diff
@@ -31,18 +31,31 @@ function remove(requestToRemove: Request) {
      * We only remove the first matching request because the order of requests matters.
      * If we were to remove all matching requests, we can end up with a final state that is different than what the user intended.
      */
-    const requests = [...persistedRequests];
```

Comment:
Right, these changes are a bit of scope creep, but it's better now imo because:

1. It avoids an unnecessary array copy
1. It only does an `Onyx.set` if the request to remove is actually found.

Reply:
Cool - I agree with that reasoning just double checking :D

Reply:
This actually caused a [regression](https://github.com/Expensify/App/issues/28825). Mutating the `persistedRequests` directly caused to the networks queue to get stuck.

Reply:
Hmm interesting. Let's continue the convo somewhere?

Reply:
https://expensify.slack.com/archives/C01GTK53T8Q/p1696433358014909